### DISCUSSION TITLE
Question about how to access the Redirect function from the @shopify/app-bridge/actions

### DISCUSSION THREAD

**Discussion by john-kuo** [kinngh/shopify-node-express-mongodb-app#121] [Q&A] [answered]
hey @kinngh , sorry to bother you again,

As you know Shopify doesn't allow a redirect call inside a i-frame, and so the only way to redirect to a new page is by using their [Redirect](https://shopify.dev/docs/api/app-bridge/previous-versions/actions/navigation/redirect-navigate#plain-javascript) function from the app-bridge package.

I looked over the project code and the place that imported the app-bridge.js package 

`   <script src="https://cdn.shopify.com/shopifycloud/app-bridge.js?apiKey=%VITE_SHOPIFY_API_KEY%"></script>
`
 is inside the **index.html** in the client folder, but can't quite figure out how to integrate the Redirect function into the React components...

Could you please help me out here?

**Comment by kinngh** [comment, marked as answer]
2 things.

1. If you pulled the package as it stands today, you'd notice I'm redirecting to a new page in a new tab using `open("url", "_blank")`. See [here](https://github.com/kinngh/shopify-node-express-mongodb-app/blob/06fec5fbcf6f83b2733e6be512dc38ce8b5e916b/client/pages/Index.jsx#L57).
2. AppBridge CDN now heavily relies on more native functions and you can read about it [here](https://shopify.dev/docs/api/app-bridge-library/apis/navigation)

**Reply by john-kuo** [reply]
@kinngh thanks, man.  The open method worked.  I just became your sponsor :)

**Reply by kinngh** [reply]
Thank you so much for the sponsor! Really appreciate it! :)
